src/base/misc/disks.c

```diff
@@ -538,9 +538,17 @@ static void image_auto(struct disk *dp)
     dp->num_secs = (unsigned long long)dp->tracks * dp->heads * dp->sectors;
   } else if (sect[510] == 0x55 && sect[511] == 0xaa) {
     filesize = lseek64(dp->fdesc, 0, SEEK_END);
+    if (filesize & (SECTOR_SIZE - 1) ) {
+      error("Access to partial last sector (%u bytes) is truncated!",
+	    filesize & (SECTOR_SIZE - 1) );
+    }
     dp->num_secs = (filesize /* + SECTOR_SIZE - 1 */ ) / SECTOR_SIZE;
     dp->heads = 255;
     dp->sectors = 63;
+    if (dp->num_secs % (dp->heads * dp->sectors) ) {
+      error("Access to partial last cylinder (%u sectors) is truncated!",
```

Comment:
I think the better wording is:
`hdimage size is not cylinder-aligned (%llu sectors), truncated!`

Reply:
I mean I tried to amend your patch-set
by hands, but there are just too many
warnings, including the one of nested
comments.